driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/mkdirat.bpf.c

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2022 The Falco Authors.
+ *
+ * This file is dual licensed under either the MIT or GPL 2. See MIT.txt
+ * or GPL2.txt for full copies of the license.
+ */
+
+#include <helpers/interfaces/fixed_size_event.h>
+#include <helpers/interfaces/variable_size_event.h>
+
+/*=============================== ENTER EVENT ===========================*/
+
+SEC("tp_btf/sys_enter")
+int BPF_PROG(mkdirat_e,
+	     struct pt_regs *regs,
+	     long id)
+{
+	struct ringbuf_struct ringbuf;
+	if(!ringbuf__reserve_space(&ringbuf, MKDIRAT_E_SIZE))
+	{
+		return 0;
+	}
+
+	ringbuf__store_event_header(&ringbuf, PPME_SYSCALL_MKDIRAT_E, MKDIRAT_E_SIZE);
+
+	/*=============================== COLLECT PARAMETERS  ===========================*/
+
+	// Here we have no parameters to collect.
+
+	/*=============================== COLLECT PARAMETERS  ===========================*/
+
+	ringbuf__submit_event(&ringbuf);
+
+	return 0;
+}
+
+/*=============================== ENTER EVENT ===========================*/
+
+/*=============================== EXIT EVENT ===========================*/
+
+SEC("tp_btf/sys_exit")
+int BPF_PROG(mkdirat_x,
+	     struct pt_regs *regs,
+	     long ret)
+{
+	struct auxiliary_map *auxmap = auxmap__get();
+	if(!auxmap)
+	{
+		return 0;
+	}
+
+	auxmap__preload_event_header(auxmap, PPME_SYSCALL_MKDIRAT_X);
+
+	/*=============================== COLLECT PARAMETERS  ===========================*/
+
+	/* Parameter 1: res (type: PT_ERRNO) */
+	auxmap__store_s64_param(auxmap, ret);
+
+	/* Parameter 2: dirfd (type: PT_FD) */
+	s32 dirfd = (s32)extract__syscall_argument(regs, 0);
+	if(dirfd == AT_FDCWD)
+	{
+		dirfd = PPM_AT_FDCWD;
+	}
+	auxmap__store_s64_param(auxmap, (s64)dirfd);
+
+	/* Parameter 3: path (type: PT_FSRELPATH) */
+	unsigned long path_pointer = extract__syscall_argument(regs, 1);
+	auxmap__store_charbuf_param(auxmap, path_pointer, USER);
+
+	/* Parameter 4: mode (type: PT_UINT32) */
+	u32 mode = (u32)extract__syscall_argument(regs, 2);
+	auxmap__store_u32_param(auxmap, mode);
```

Comment:
I wondered why `chmod_mode_to_scap` is not used here? Looking at the `chmod`, the event does not use the `PT_MODE` and, I guess, it is the event API then to push w/o conversion here.

Reply:
Uhm sorry but I didn't catch it :thinking: IIRC `mkdirat` has different modes with respect to `chmod`, something like:

```
 S_IFMT     0170000   bit mask for the file type bit field
 S_IFSOCK   0140000   socket
 S_IFLNK    0120000   symbolic link
 S_IFREG    0100000   regular file
 S_IFBLK    0060000   block device
 S_IFDIR    0040000   directory
 S_IFCHR    0020000   character device
 S_IFIFO    0010000   FIFO
```

so i didn't understand why we need to use the same `chmod_mode_to_scap` helper, or maybe i didn't catch your question :thinking:

Reply:
Ohh sorry! The answer is absolutely yes! We need to add some helpers to do that and as you will see this is not the only case (`renameat2` flags for example) :( I will add it to our tracking issue https://github.com/falcosecurity/libs/issues/515

Reply:
Already have seen the Todo there :) Thanks.